src/util/std_expr.h

```diff
@@ -4460,4 +4460,78 @@ inline cond_exprt &to_cond_expr(exprt &expr)
   return ret;
 }
 
+/// \brief Construct an array from an anonymous function
```

Comment:
It's good that there is a comment here, but it isn't enough that I can figure out what it does. Why is it always an array? Is it because it's used in flattening?

Reply:
Fixed in #4672, which now takes care of the initial commits in this PR.

Reply:
Thanks, I find that much clearer